### PR TITLE
Echo client (over IP) in Android demo app.

### LINKED ISSUE #1104: Add support for Echo Client in Android App
Using the information in the QR code the Android app should be able to connect to the device and implement an echo client. The accessory will act as the echo server.

### LINKED ISSUE #1450: Provide CHIPDeviceConnection wrapper in JNI layer.

### PR DESCRIPTION
**Pull request by vidhis88**
1. Adding the JNI bridge for ChipDeviceController
2. Adding a screen to take in the IP address of the CHIP device and 
message to echo

Demo: 
https://drive.google.com/file/d/15jIrB8Cz6k9MNixhhdeJEAWj6fmNCAJR/view

fixes #1104 and #1450

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/controller/CHIPDeviceController.h [hunk 1]
```diff
@@ -60,6 +60,7 @@ class DLL_EXPORT ChipDeviceController : public SecureSessionMgrCallback
     void * AppState;
 
     CHIP_ERROR Init(NodeId localDeviceId);
+    CHIP_ERROR Init(NodeId localDeviceId, System::Layer * systemLayer, InetLayer * inetLayer);
```
Comment (andy31415) [inline src/controller/CHIPDeviceController.h:63]: some doc comments on when one Init should be used vs another would be helpful.

Is the first version useful if CONFIG_DEVICE_LAYER is not defined? How aboud ifdef-ing which version of init is available based compile flags?

Comment (vidhis88) [reply src/controller/CHIPDeviceController.h:63]: Added a ticket to track unifying the Init functions into a single meaningful one -- #1902.

Comment (vidhis88) [reply src/controller/CHIPDeviceController.h:70]: Added docs as well.

**Comment by woody-apple** [comment]
@saurabhst @jelderton @BroderickCarlin ?

**Review by saurabhst — APPROVED** [review]

File: config/android/CHIPProjectConfig.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Project-specific configuration file for iOS builds.
```
Comment (jelderton) [inline config/android/CHIPProjectConfig.h:23]: I realize this is a copy from ios, but since it's not a link I would fix this comment to say Android

Comment (vidhis88) [reply config/android/CHIPProjectConfig.h]: Good catch. Thanks, updated.

**Review by jelderton — APPROVED** [review]

**Pull request merged**